### PR TITLE
[core] Simplified logs output

### PR DESCRIPTION
**Pull request by biodranik**
Example:

```
I(1) 0.72366 search/processor.cpp:210 SetPreferredLocale(): New preferred locale: en
D(1) 0.72374 search/processor.cpp:229 SetInputLocale(): New input locale: en locale code:
W(2) 0.72979 search/geocoder.cpp:736 CacheWorldLocalities(): Can't find World map file.
```

Before change:
```
LOG TID(1) DEBUG  7.1459e-05 platform/platform_mac.mm:135 Platform: Resources Directory: /
LOG TID(1) DEBUG 0.000118709 platform/platform_mac.mm:136 Platform: Writable Directory: /
LOG TID(1) DEBUG    0.000133 platform/platform_mac.mm:137 Platform: Tmp Directory: /
LOG TID(1) DEBUG 0.000136417 platform/platform_mac.mm:138 Platform: Settings Directory: /
LOG TID(1) INFO  0.000162875 qt/main.cpp:121 main(): Organic Maps 2024.04.22-3-c91add3a-Darwin started, detected CPU cores: 1024
```

### CODE REVIEW

**Comment by Tmpod** [comment]
I was actually thinking of doing something like this myself the other day, when I was looking around in logcat. This is nice!

**Comment by biodranik** [comment]
@vng ⬆️

File: base/logging.cpp [hunk 1]
```diff
@@ -12,37 +12,33 @@
...
-  return names[level];
+  return std::string{} + names[level];
 }
 
-bool FromString(std::string const & s, LogLevel & level)
+std::optional<LogLevel> FromString(std::string const & s)
 {
   auto const & names = GetLogLevelNames();
-  auto it = find(names.begin(), names.end(), s);
+  const auto it = std::find(names.begin(), names.end(), std::toupper(s[0]));
```
Comment (vng) [inline base/logging.cpp:33]: ASSERT(!s.empty(), ());

File: base/logging.cpp [hunk 2]
```diff
@@ -12,37 +12,33 @@
 #include <mutex>
 #include <sstream>
 
+namespace base
+{
 namespace
 {
 std::mutex g_logMutex;
+std::array<char, NUM_LOG_LEVELS> constexpr kLogLevelNames = {{'D', 'I', 'W', 'E', 'C'}};
```
Comment (vng) [inline base/logging.cpp:20]: Keep as static var in GetLogLevelNames?

**Comment by biodranik** [comment]
@vng fixed your comments, PTAL

**Review by vng — APPROVED** [review]

**Pull request merged**